File: DataSets/DataSet-New/LeetCodeTasks - Develop/LeetCodeTasks/3105_longest-strictly-increasing-or-strictly-decreasing-subarray/solutions/java/solution.java
class Solution {
    public int longestMonotonicSubarray(int[] nums) {
        if (nums.length == 1) return 1;

        int maxLength = 1;
        int currentLength = 1;

        // Check for strictly increasing
        for (int i = 1; i < nums.length; i++) {
            if (nums[i] > nums[i - 1]) {
                currentLength++;
                maxLength = Math.max(maxLength, currentLength);
            } else {
                currentLength = 1;
            }
        }

        // Reset current length for strictly decreasing check
        currentLength = 1;

        // Check for strictly decreasing
        for (int i = 1; i < nums.length; i++) {
            if (nums[i] < nums[i - 1]) {
                currentLength++;
                maxLength = Math.max(maxLength, currentLength);
            } else {
                currentLength = 1;
            }
        }

        return maxLength;
    }
}

File: DataSets/DataSet-New/LeetCodeTasks - Develop/LeetCodeTasks/3105_longest-strictly-increasing-or-strictly-decreasing-subarray/solutions/c/solution.c
#include <stdio.h>

int longestMonotonicSubarray(int* nums, int numsSize) {
    if (numsSize == 1) {
        return 1;
    }
    
    int max_length = 1;
    int current_length = 1;
    int i;

    // Check for strictly increasing sequences
    for (i = 1; i < numsSize; i++) {
        if (nums[i] > nums[i - 1]) {
            current_length++;
            if (current_length > max_length) {
                max_length = current_length;
            }
        } else {
            current_length = 1;
        }
    }
    
    // Check for strictly decreasing sequences
    current_length = 1; // reset for new direction check
    for (i = 1; i < numsSize; i++) {
        if (nums[i] < nums[i - 1]) {
            current_length++;
            if (current_length > max_length) {
                max_length = current_length;
            }
        } else {
            current_length = 1;
        }
    }

    return max_length;
}
